apis/kueue/webhooks/clusterqueue_webhook_test.go

```diff
@@ -152,6 +146,39 @@ func TestValidateClusterQueue(t *testing.T) {
 				field.Required(specField.Child("namespaceSelector", "matchExpressions").Index(0).Child("values"), ""),
 			},
 		},
+		{
+			name: "multiple independent and codependent resources",
+			clusterQueue: testingutil.MakeClusterQueue("cluster-queue").
+				Resource(testingutil.MakeResource("cpu").
+					Flavor(testingutil.MakeFlavor("alpha", "0").Obj()).
+					Flavor(testingutil.MakeFlavor("beta", "0").Obj()).Obj()).
+				Resource(testingutil.MakeResource("memory").
+					Flavor(testingutil.MakeFlavor("alpha", "0").Obj()).
+					Flavor(testingutil.MakeFlavor("beta", "0").Obj()).Obj()).
+				Resource(testingutil.MakeResource("example.com/gpu").
+					Flavor(testingutil.MakeFlavor("gamma", "0").Obj()).
+					Flavor(testingutil.MakeFlavor("omega", "0").Obj()).Obj()).
+				Obj(),
+		},
+		{
+			name: "multiple resources with partial flavor match",
+			clusterQueue: testingutil.MakeClusterQueue("cluster-queue").
+				Resource(testingutil.MakeResource("cpu").
+					Flavor(testingutil.MakeFlavor("alpha", "0").Obj()).
+					Flavor(testingutil.MakeFlavor("beta", "0").Obj()).Obj()).
+				Resource(testingutil.MakeResource("memory").
+					Flavor(testingutil.MakeFlavor("beta", "0").Obj()).
+					Flavor(testingutil.MakeFlavor("alpha", "0").Obj()).Obj()).
+				Resource(testingutil.MakeResource("example.com/gpu").
+					Flavor(testingutil.MakeFlavor("alpha", "0").Obj()).
+					Flavor(testingutil.MakeFlavor("omega", "0").Obj()).Obj()).
+				Obj(),
+			wantErr: field.ErrorList{
+				field.Invalid(specField.Child("resources").Index(1).Child("flavors"), nil, ""),
+				field.Invalid(specField.Child("resources").Index(2).Child("flavors"), nil, ""),
+				field.Invalid(specField.Child("resources").Index(2).Child("flavors"), nil, ""),
```

Comment:
the last two errors being duplicates isn't ideal, can we compare the error messages?

Reply:
Maybe split this into two testcases, one is for the exactly same orders, another one for the validation of partly same flavors in different resources.

Reply:
I decided to split it, because the details in some errors are quite cumbersome to write and not worth testing, IMO.